### PR TITLE
Add optional chaining to startsWith method

### PR DESCRIPTION
**Pull request by jenniemc**
## Summary
This PR adds optional chaining directly to the startsWith() method inside the components that make up the CalendarWidget

### Background

Google Analytic is recording error from `/health-care/schedule-view-va-appointments/appointments/new-appointment/select-date/`.  The error is listed below and on the screenshot of GA dashboard. 

- Cannot read properties of undefined (reading 'startsWith')
- undefined is not an object (evaluating 'e.start.startsWith')

Previous PRs did not fix the error. This PR adds optional chaining directly to the startsWith() method.

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#53873

## Screen Shot of error key captured in GA

Screenshot 2023-08-01 at 11 43 09 AM

## What areas of the site does it impact?

VAOS google analytic

### CODE REVIEW

File: src/applications/vaos/new-appointment/components/DateTimeRequestPage/SelectedIndicator.jsx [hunk 1]
```diff
@@ -3,12 +3,13 @@ import moment from 'moment';
 
 export function getSelectedLabel(date, selectedDates) {
   const matchingTimes = selectedDates.filter(selected =>
-    selected.startsWith(date),
+    selected?.startsWith(date),
```
Comment (vbahinwillit) [inline src/applications/vaos/new-appointment/components/DateTimeRequestPage/SelectedIndicator.jsx:6]: I don't think you get this far if selectedDates is empty. So no need for the '?'.

File: src/applications/vaos/new-appointment/components/DateTimeRequestPage/SelectedIndicator.jsx [hunk 2]
```diff
@@ -18,7 +19,7 @@ export function getSelectedLabel(date, selectedDates) {
 export default function SelectedIndicator({ date, selectedDates }) {
   const bubbles = selectedDates
     .reduce((selectedFieldValues, currentDate) => {
-      if (currentDate.startsWith(date)) {
+      if (currentDate?.startsWith(date)) {
```
Comment (vbahinwillit) [inline src/applications/vaos/new-appointment/components/DateTimeRequestPage/SelectedIndicator.jsx:22]: Same as above.

File: src/applications/vaos/new-appointment/components/DateTimeRequestPage/SelectedIndicator.jsx (hunk 1 above)
Comment (jenniemc) [reply src/applications/vaos/new-appointment/components/DateTimeRequestPage/SelectedIndicator.jsx:6]: Done. Removed the '?'

File: src/applications/vaos/new-appointment/components/DateTimeRequestPage/SelectedIndicator.jsx (hunk 2 above)
Comment (jenniemc) [reply src/applications/vaos/new-appointment/components/DateTimeRequestPage/SelectedIndicator.jsx:22]: Done. Removed the '?'

**Review by vbahinwillit — APPROVED** [review]

**Pull request merged**